### PR TITLE
Add delegatesFocus in the Shadow DOM spec.

### PR DESCRIPTION
**Pull request by TakayoshiKochi**
Original proposal:
https://github.com/w3c/webcomponents/blob/gh-pages/proposals/ShadowRoot-delegatesFocus-Proposal.md

Discussion:
https://github.com/w3c/webcomponents/issues/126

### CODE REVIEW

**Comment by TakayoshiKochi** [comment]
@hayatoito PTAL this is continuation from #455.

File: spec/shadow/index.html [hunk 1]
```diff
@@ -1137,6 +1137,20 @@
       </section>
 
       <section>
+        <h3>Focus</h3>
+      <p>A <a>shadow host</a> can delegate focus to its <a>shadow root</a> by assigning a boolean <a href="#widl-ShadowRootInit-delegatesFocus">delegatesFocus</a> flag to be true in <a href="#idl-def-ShadowRootInit">ShadowRootInit</a> dictionary. If omitted, a <a>shadow host</a> does not delegate focus to its shadow root, and the shadow host itself can be focusable.</p>
```
Comment (hayatoito) [inline spec/shadow/index.html:1141]: Could you surround "delegate focus" by `<dfn></dfn>`? And reference it in other places.

File: spec/shadow/index.html [hunk 2]
```diff
@@ -1137,6 +1137,20 @@
...
+        <p>When a shadow host <var>HOST</var> delegates focus, user agent <strong>must</strong> behave as follows.
+          <ol>
+            <li>In <a>sequential focus navigation</a>, <var>HOST</var> itself will be skipped. See the next secition for the formal definition.</li>
+            <li>When <var>HOST</var> is focused by <code>focus()</code> method or <code>autofocus</code> attribute: The first <a>focusable area</a> in the tree order in <var>HOST</var>'s <a>shadow tree</a> gets focus.</li>
```
Comment (hayatoito) [inline spec/shadow/index.html]: Tabindex is not used here to determine the first focusable area?

Comment (TakayoshiKochi) [inline spec/shadow/index.html]: no, the first one in the focus navigation order of its shadow root.
Replaced the sentence.

File: spec/shadow/index.html (hunk 1 above)
Comment (TakayoshiKochi) [inline spec/shadow/index.html:1141]: Done.

**Comment by TakayoshiKochi** [comment]
@hayatoito PTAL

**Pull request merged**